main.py

```diff
@@ -163,9 +167,33 @@ def handle_errors(self, event):
         The event is sent only if the error is related to a request made
         by flow_manager.
         """
-        xid = event.content["message"].header.xid.value
-        error_type = event.content["message"].error_type
-        error_code = event.content["message"].code
+        message = event.content["message"]
+        connection = event.source
+        switch = connection.switch
+
+        error_data = message.data.pack()
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if (message.code == BadActionCode01.OFPBAC_BAD_OUT_PORT or
+           message.code == BadActionCode04.OFPBAC_BAD_OUT_PORT):
+
+            for action in error_packet.actions:
+                try:
+                    iface = switch.get_interface_by_port_no(action.port.value)
+                except AttributeError:
+                    iface = switch.get_interface_by_port_no(action.port)
+
+                # Check interface to drop packets forwarded to it
+                if iface:
+                    if connection.protocol.version == 0x01:
+                        iface.config = PortConfig01.OFPPC_NO_FWD
+                    elif connection.protocol.version == 0x04:
+                        iface.config = PortConfig04.OFPPC_NO_FWD
```

Comment:
Same thing here. There should be a function somewhere that returned the right object for any `pyof` supported version.

Reply:
Something like this, just to have an idea of what I'm talking about:

```
port_config = pyof.get_port_config_for_version(connection.protocol.version)
iface.config = port_config.OFPPC_NO_FWD
```